src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs

```diff
@@ -29,6 +30,10 @@ public void Start()
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                // Set a new cancellation token source which will be used to cancel this task in case
+                // new loading task starts or manager ui is closed while loading packages.
+                _packageManagerControl._loadCts = new CancellationTokenSource();
```

Comment:
Do you need to dispose the previous source? See: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#always-dispose-cancellationtokensources-used-for-timeouts

Reply:
I just moved this one from the PackageManagerControl.  

I thought it was fine since Disposing the control causes cleanup, but actually we've always been improperly handling this because every search creates a new token. I think yes...we should have been disposing those?  Please clarify if you agree with that, or it's specific to the move I made.

Reply:
Yup I agree with that.